src/relation/transforms.rs

```diff
@@ -216,6 +216,24 @@ impl Reduce {
     pub fn rename_fields<F: Fn(&str, &Expr) -> String>(self, f: F) -> Reduce {
         Relation::reduce().rename_with(self, f).build()
     }
+
+    /// In the current `Reduce` contains grouping columns,
+    /// adds them to the `aggregate` field.
+    pub fn push_grouping_columns(self) -> Reduce {
+        if self.group_by().is_empty() {
+            self
+        } else {
+            Reduce::builder()
+                .with(self.clone())
```

Comment:
if you capture self, you should use self here.
Collect grouping columns before so that you do not try to borrow from a moved value.